Assets/Scripts/Framework/Train/TrainController.cs

```diff
@@ -0,0 +1,271 @@
+using UnityEngine.Splines;
+using FishNet.Object;
+using UnityEngine;
+using FishNet;
+using System;
+
+namespace DerailedDeliveries.Framework.Train
+{
+    /// <summary>
+    /// Class responsible for moving train along rails spline.
+    /// </summary>
+    [RequireComponent(typeof(TrainEngine))]
+    public class TrainController : NetworkBehaviour
+    {
+        /// <summary>
+        /// Reference to the spline line data.
+        /// </summary>
+        [field: SerializeField]
+        public SplineContainer Spline { get; private set; }
+        
+        [Header("Train Config")]
+        [SerializeField, Range(0, 1)]
+        private float _trainFrontStartTime = 0f;
+
+        [SerializeField]
+        private float _heightOffset = 1.0f;
+
+        [Header("Wagons Config")]
+        [SerializeField]
+        private Transform _frontWagon = null;
+        
+        [SerializeField]
+        private float _wagonFollowDistance = 0f;
+
+        [SerializeField]
+        private float _wagonSpacing = 0f;
+
+        [SerializeField]
+        private Transform[] _wagons = null;
```

Comment:
These default values also are not needed.
```suggestion
        [Header("Wagons Config")]
        [SerializeField]
        private Transform _frontWagon;
        
        [SerializeField]
        private float _wagonFollowDistance;

        [SerializeField]
        private float _wagonSpacing;

        [SerializeField]
        private Transform[] _wagons;
```